<launch>
    <param name="/use_sim_time" value="true" />

    <param name="sampling_round" value="0" />
    <param name="sampling_num" value="150" />
    <param name="iter_cycle" value="5" />
    <param name="gap_limit" value="0.0" />
    <param name="angle_limit" value="0.00" />
    <param name="residual_limit" value="5" />
    <param name="slerp" value="200" />

    <param name="scan_line" type="int" value="16" />

    <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automaticlly -->
    <param name="mapping_skip_frame" type="int" value="1" />

    <!-- remove too closed points -->
    <param name="minimum_range" type="double" value="0.3"/>


    <param name="mapping_line_resolution" type="double" value="0.2"/>
    <param name="mapping_plane_resolution" type="double" value="0.4"/>

    <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="log" />

    <node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="log" />

    <node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="log" />
    
    <node pkg="aloam_velodyne" type="icpTrajectory" name="icpTrajectory" output="screen" />

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/aloam_velodyne.rviz" />
    </group>

</launch>
